doc/developer-guides/coding_guidelines.rst

```diff
@@ -1281,6 +1749,68 @@ Compliant example::
        showcase_u32_3 = showcase_u32_1 * showcase_u32_2 >> 4U;
 
 
+EP-06: Overflow shall not be allowed
+====================================
+
+Compliant example::
+
+    uint8_t showcase = 255U;
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       uint8_t showcase = 255U + 1U;
+
+
+EP-07: Negation shall not be performed on unsigned expression
+=============================================================
+
+Compliant example::
+
+    int32_t showcase = -10;
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       int32_t showcase = -10U;
+
+
+EP-08: A pointer shall not be assigned to another pointer whose lifetime is longer
```

Comment:
Seems like the requirement is on the value of the pointer variable rather than the lifetime of the pointer variable. Your non-compliant example below is storing the address of a local variable into a more global pointer variable.  It's not assigning one pointer to another as your stated requirement says.

Reply:
@dbkinder 
A pointer is a variable whose value is the address of another variable.
In the non-compliant example, the address of a local variable is a local pointer and is assigned to a global variable.
I will add a variable `uint32_t *showcase_ptr_local` to make the example more straightforward.

Reply:
I still think the requirement needs to talk about the value of the pointer, not the lifetime of the pointer variable.  As written, I can't assign the address of a global variable to a local pointer and then assign this pointer value to a global pointer (a perfectly valid thing to do).

Is this a compliant example:
```
      uint32_t *showcase_ptr_global;
      uint32_t showcase_global = 100U;
       
       void func_showcase(void)
       {
           uint32_t *showcase_ptr_local;
 
           showcase_ptr_local = &showcase_global;
           showcase_ptr_global = showcase_ptr_local;   
       } 
       void main(void)
       {
           func_showcase();
           printf("*showcase_ptr_global %d \n", *showcase_ptr_global);
       }  
```